sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -41,29 +41,29 @@ object Klaviyo {
     private val preInitQueue: Queue<Operation<Unit>> = LinkedList()
 
     /**
-     * Since the analytics module owns these services, it must register them.
+     * This method is provided for apps that are unable to register their API key immediately
+     * on app launch in order enable limited SDK functionality including tracking app lifecycle,
+     * automated push token collection, and handling universal tracking links.
      *
-     * This registration is a lambda invoked when the service is required, not instantiated now
-     */
-    private fun initializeServices() = Registry.apply {
-        registerOnce<ApiClient> { KlaviyoApiClient }
-        registerOnce<State> {
-            KlaviyoState().also { state ->
-                register<StateSideEffects>(StateSideEffects(state))
-            }
-        }
-    }
-
-    /**
-     * Use this method to register Klaviyo for lifecycle functions. This is necessary for
-     * apps that are not able to [initialize] Klaviyo immediately on app launch, but would like to
-     * utilize Klaviyo Forms
+     * Your API key still must be provided as early as possible for full SDK functionality!
      *
      * @param applicationContext
      */
     fun registerForLifecycleCallbacks(applicationContext: Context) = safeApply {
-        val application = applicationContext.applicationContext as? Application
-        application?.apply {
+        if (!Registry.isRegistered<Config>()) {
+            // Register a partial config, missing API Key, to allow lifecycle tracking and context access for partial functionality
+            Registry.register<Config>(
+                Registry.configBuilder
+                    .applicationContext(applicationContext)
+                    .build()
+            )
+        }
+
+        // Some APIs (such as deep linking) work without an API key, so we can register the core service now
+        Registry.registerOnce<ApiClient> { KlaviyoApiClient }
```

Comment:
and API requests that do require an API key are all still behind the `Klaviyo` facade, and will have the same behavior as before if you call them before properly initializing.